crates/binstalk/src/ops/resolve.rs

```diff
@@ -91,45 +92,72 @@ async fn resolve_inner(
         .collect::<Result<Vec<_>, _>>()?;
     let resolvers = &opts.resolvers;
 
-    let mut handles: Vec<(Arc<dyn Fetcher>, _)> =
-        Vec::with_capacity(desired_targets.len() * resolvers.len());
-
-    let data = Arc::new(Data::new(
-        package_info.name.clone(),
-        package_info.version_str.clone(),
-        package_info.repo.clone(),
-    ));
+    let binary_name = match package_info.binaries.as_slice() {
+        [bin] if bin.name != package_info.name => Some(CompactString::from(bin.name.as_str())),
+        _ => None,
+    };
 
-    handles.extend(
-        resolvers
-            .iter()
-            .cartesian_product(desired_targets.into_iter().map(|(triple, target)| {
-                debug!("Building metadata for target: {target}");
+    let mut handles: Vec<(Arc<dyn Fetcher>, _)> = Vec::with_capacity(
+        desired_targets.len() * resolvers.len() * if binary_name.is_some() { 2 } else { 1 },
```

Comment:
```suggestion
        desired_targets.len() * resolvers.len() + if binary_name.is_some() { desired_targets.len() } else { 0 },
```

Reply:
nice catch with the over allocation! Thinking about it, should there be parenthesis?
```rust
desired_targets.len() * (resolvers.len() + binary_name.as_ref().map(|_|1).unwrap_or_default())
```